code/src/layouts/modals/auth/components/sign-in-form/SignInForm.test.tsx

```diff
@@ -0,0 +1,84 @@
+import { screen, fireEvent, waitFor, act } from "@testing-library/react";
+import SignInForm from "@/layouts/modals/auth/components/sign-in-form/SignInForm";
+import useSignIn from "@/hooks/use-sign-in/useSignIn";
+import { renderWithProviders } from "@/utils/test-utils";
+
+jest.mock("@/hooks/use-sign-in/useSignIn", () => ({
+  __esModule: true,
+  default: jest.fn()
+}));
+
+const mockSignIn = jest.fn();
+(useSignIn as jest.Mock).mockReturnValue([mockSignIn, { isLoading: false }]);
+
+const mockFormValues = {
+  email: "test@example.com",
+  password: "Helloworld123!"
+};
+
+describe("SignInForm", () => {
+  beforeEach(() => {
+    renderWithProviders(<SignInForm />);
+  });
+
+  test("renders the form correctly", () => {
+    const passwordInput = screen.getByLabelText(/signIn.password.field/i);
+    const emailInput = screen.getByLabelText(/signIn.email.field/i);
+    const submitButton = screen.getByRole("button", { name: /signIn.button/i });
+
+    expect(passwordInput).toBeInTheDocument();
+    expect(emailInput).toBeInTheDocument();
+    expect(submitButton).toBeInTheDocument();
+  });
+
+  test("handles input changes and form submission", async () => {
+    const emailInput = screen.getByLabelText(/signIn.email.field/);
+    const passwordInput = screen.getByLabelText(/signIn.password.field/i);
+    const submitButton = screen.getByRole("button", { name: /signIn.button/i });
+
+    await act(async () => {
+      fireEvent.change(emailInput, { target: { value: mockFormValues.email } });
+      fireEvent.change(passwordInput, {
+        target: { value: mockFormValues.password }
+      });
+    });
+
+    await waitFor(() => {
+      expect(emailInput).toHaveValue(mockFormValues.email);
+      expect(passwordInput).toHaveValue(mockFormValues.password);
+    });
+
+    fireEvent.click(submitButton);
+
+    await waitFor(() => {
+      expect(mockSignIn).toHaveBeenCalledWith(mockFormValues);
+    });
+  });
+
+  test("displays validation errors", async () => {
+    const submitButton = screen.getByRole("button", { name: /signIn.button/i });
+
+    fireEvent.click(submitButton);
+
+    await waitFor(() => {
+      const emailErrorMessage = screen.getByText(
+        "Please provide a valid email address"
+      );
+      const passwordErrorMessage = screen.getByText(
+        "Password must be at least 8 characters long"
+      );
+
+      expect(emailErrorMessage).toBeInTheDocument();
+      expect(passwordErrorMessage).toBeInTheDocument();
+    });
```

Comment:
tests also works with related a simple text for it without translations. Just text will be enough

Reply:
I'll create issue for adding translations, and refactor this test